test/runner/lib/cloud/vcenter.py

```diff
@@ -62,10 +79,16 @@ def setup(self):
         """Setup the cloud resource before delegation and register a cleanup callback."""
         super(VcenterProvider, self).setup()
 
-        if self._use_static_config():
-            self._setup_static()
-        else:
-            self._setup_dynamic()
+        # TODO mostly thinking out loud here, to try out the idea
+        if self.vmware_test_platform is 'govcsim' or None:
+            if self._use_static_config():
+                self._setup_static_simulator()
```

Comment:
Since there's no static simulator option, it would be convenient to keep the static (bare metal) auto-detection support. If the config file is present, default to static bare metal. If it's not, default to dynamic via govcsim.

The env var would only be needed to force dynamic in the presence of the static config file, or to change the default dynamic provisioning mode.